src/gax/src/retry_policy.rs

```diff
@@ -974,42 +1025,40 @@ mod tests {
 
     #[test]
     fn test_limited_attempt_count_on_throttle_none() {
-        let mut mock = MockPolicy::new();
-        mock.expect_on_throttle().times(1..).returning(|_, _| None);
-
-        let now = std::time::Instant::now();
-        let policy = LimitedAttemptCount::custom(mock, 3);
-        assert!(policy.on_throttle(now, 1).is_none());
-        assert!(policy.on_throttle(now, 2).is_none());
-        assert!(policy.on_throttle(now, 3).is_some());
-    }
-
-    #[test]
-    fn test_limited_attempt_count_on_throttle_some() {
         let mut mock = MockPolicy::new();
         mock.expect_on_throttle()
             .times(1..)
-            .returning(|_, _| Some(transient_error()));
+            .returning(|_, _, e| ThrottleResult::Continue(e));
 
         let now = std::time::Instant::now();
         let policy = LimitedAttemptCount::custom(mock, 3);
-        assert!(policy.on_throttle(now, 1).is_some());
-        assert!(policy.on_throttle(now, 2).is_some());
-        assert!(policy.on_throttle(now, 3).is_some());
+        assert!(matches!(
+            policy.on_throttle(now, 1, unavailable()),
+            ThrottleResult::Continue(_)
+        ));
+        assert!(matches!(
+            policy.on_throttle(now, 2, unavailable()),
+            ThrottleResult::Continue(_)
+        ));
+        assert!(matches!(
+            policy.on_throttle(now, 3, unavailable()),
```

Comment:
this doesn't happen in reality, so maybe just test a single `on_throttle` returning `Continue`?